app/models/service_order.rb

```diff
@@ -34,6 +39,12 @@ def checkout
     update_attributes(:state => STATE_ORDERED)
   end
 
+  def clear
+    raise "Invalid operation [clear] for Service Order in state [#{state}]" if state == STATE_ORDERED
```

Comment:
Can you add a method for `ordered?`?  I think it would make the if conditions cleaner